### ISSUE TITLE
unarchive module error--cloud not support Chinese

### ISSUE THREAD

**Issue by zhijieqing** [ansible/ansible-modules-core#4478]
##### ISSUE TYPE

- Bug Report
##### COMPONENT NAME

unarchive
##### ANSIBLE VERSION

```
ansible 2.1.1.0
```
##### OS / ENVIRONMENT

CentOS Linux release 7.1.1503 (Core)
##### SUMMARY

Thank you for your reply!
But this issue [[](https://github.com/ansible/ansible/issues/17146) ] also appear in version 2.1.

the file [unarchive_test.zip] is in attach files and archive with [tar -czvf] .

Thank you.

[unarchive_test.zip.tar.gz](https://github.com/ansible/ansible-modules-core/files/428124/unarchive_test.zip.tar.gz)
##### STEPS TO REPRODUCE

```

```

##### EXPECTED RESULTS

[root@localhost ~]# unzip unarchive_test.zip -d /tmp2/
Archive:  unarchive_test.zip
   creating: /tmp2/unarchive_test/
 extracting: /tmp2/unarchive_test/??????.txt  
[root@localhost ~]# ll /tmp2/unarchive_test/
总用量 4
-rw-r--r--. 1 root root 30 8月  20 12:01 新建文本文档.txt
##### ACTUAL RESULTS

```
[root@localhost ~]# ansible 127.0.0.1 -m unarchive -a "src=unarchive_test.zip dest=/tmp"
127.0.0.1 | FAILED! => {
    "changed": false, 
    "failed": true, 
    "msg": "path /tmp/unarchive_test/新建文本文档.txt does not exist", 
    "path": "/tmp/unarchive_test/新建文本文档.txt", 
    "state": "absent"
}

[root@localhost ~]# ll /tmp/unarchive_test/
总用量 4
-rw-r--r--. 1 root root 30 8月  20 12:01 #U65b0#U5efa#U6587#U672c#U6587#U6863.txt

[root@localhost ~]# ansible 127.0.0.1 -m unarchive -a "src=unarchive_test.zip dest=/tmp" -vvvv
Using /etc/ansible/ansible.cfg as config file
Loaded callback minimal of type stdout, v2.0
<127.0.0.1> ESTABLISH SSH CONNECTION FOR USER: None
<127.0.0.1> SSH: EXEC ssh -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey 
-o PasswordAuthentication=no -o ConnectTimeout=10 -o ControlPath=/root/.ansible/cp/ansible-ssh-%h-%p-%r 127.0.0.1 '/bin/sh -c '"'"'( umask 77 && mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1471665795.29-86900711158190 `" && echo ansible-tmp-1471665795.29-86900711158190="` echo $HOME/.ansible/tmp/ansible-tmp-1471665795.29-86900711158190 `" ) && sleep 0'"'"''<127.0.0.1> ESTABLISH SSH CONNECTION FOR USER: None
<127.0.0.1> SSH: EXEC ssh -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey 
-o PasswordAuthentication=no -o ConnectTimeout=10 -o ControlPath=/root/.ansible/cp/ansible-ssh-%h-%p-%r 127.0.0.1 '/bin/sh -c '"'"'( umask 77 && mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1471665795.64-22894972601203 `" && echo ansible-tmp-1471665795.64-22894972601203="` echo $HOME/.ansible/tmp/ansible-tmp-1471665795.64-22894972601203 `" ) && sleep 0'"'"''<127.0.0.1> PUT /tmp/tmpcwZsHb TO /root/.ansible/tmp/ansible-tmp-1471665795.64-22894972601203/stat
<127.0.0.1> SSH: EXEC sftp -b - -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publ
ickey -o PasswordAuthentication=no -o ConnectTimeout=10 -o ControlPath=/root/.ansible/cp/ansible-ssh-%h-%p-%r '[127.0.0.1]'<127.0.0.1> ESTABLISH SSH CONNECTION FOR USER: None
<127.0.0.1> SSH: EXEC ssh -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey 
-o PasswordAuthentication=no -o ConnectTimeout=10 -o ControlPath=/root/.ansible/cp/ansible-ssh-%h-%p-%r -tt 127.0.0.1 '/bin/sh -c '"'"'LANG=zh_CN.UTF-8 LC_ALL=zh_CN.UTF-8 LC_MESSAGES=zh_CN.UTF-8 /usr/bin/python /root/.ansible/tmp/ansible-tmp-1471665795.64-22894972601203/stat; rm -rf "/root/.ansible/tmp/ansible-tmp-1471665795.64-22894972601203/" > /dev/null 2>&1 && sleep 0'"'"''<127.0.0.1> PUT /root/unarchive_test.zip TO /root/.ansible/tmp/ansible-tmp-1471665795.29-86900711158190/source
<127.0.0.1> SSH: EXEC sftp -b - -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publ
ickey -o PasswordAuthentication=no -o ConnectTimeout=10 -o ControlPath=/root/.ansible/cp/ansible-ssh-%h-%p-%r '[127.0.0.1]'<127.0.0.1> ESTABLISH SSH CONNECTION FOR USER: None
<127.0.0.1> SSH: EXEC ssh -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey 
-o PasswordAuthentication=no -o ConnectTimeout=10 -o ControlPath=/root/.ansible/cp/ansible-ssh-%h-%p-%r 127.0.0.1 '/bin/sh -c '"'"'( umask 77 && mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1471665796.23-250723666945300 `" && echo ansible-tmp-1471665796.23-250723666945300="` echo $HOME/.ansible/tmp/ansible-tmp-1471665796.23-250723666945300 `" ) && sleep 0'"'"''<127.0.0.1> PUT /tmp/tmpPFE8Kh TO /root/.ansible/tmp/ansible-tmp-1471665796.23-250723666945300/unarchive
<127.0.0.1> SSH: EXEC sftp -b - -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publ
ickey -o PasswordAuthentication=no -o ConnectTimeout=10 -o ControlPath=/root/.ansible/cp/ansible-ssh-%h-%p-%r '[127.0.0.1]'<127.0.0.1> ESTABLISH SSH CONNECTION FOR USER: None
<127.0.0.1> SSH: EXEC ssh -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey 
-o PasswordAuthentication=no -o ConnectTimeout=10 -o ControlPath=/root/.ansible/cp/ansible-ssh-%h-%p-%r -tt 127.0.0.1 '/bin/sh -c '"'"'LANG=zh_CN.UTF-8 LC_ALL=zh_CN.UTF-8 LC_MESSAGES=zh_CN.UTF-8 /usr/bin/python /root/.ansible/tmp/ansible-tmp-1471665796.23-250723666945300/unarchive; rm -rf "/root/.ansible/tmp/ansible-tmp-1471665796.23-250723666945300/" > /dev/null 2>&1 && sleep 0'"'"''<127.0.0.1> ESTABLISH SSH CONNECTION FOR USER: None
<127.0.0.1> SSH: EXEC ssh -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey 
-o PasswordAuthentication=no -o ConnectTimeout=10 -o ControlPath=/root/.ansible/cp/ansible-ssh-%h-%p-%r 127.0.0.1 '/bin/sh -c '"'"'rm -f -r /root/.ansible/tmp/ansible-tmp-1471665795.29-86900711158190/ > /dev/null 2>&1 && sleep 0'"'"''127.0.0.1 | FAILED! => {
    "changed": false, 
    "failed": true, 
    "invocation": {
        "module_args": {
            "dest": "/tmp", 
            "src": "unarchive_test.zip"
        }, 
        "module_name": "unarchive"
    }, 
    "module_stderr": "OpenSSH_6.6.1, OpenSSL 1.0.1e-fips 11 Feb 2013\r\ndebug1: Reading configuration data /etc/ssh/ssh_config\r\ndebug1: /etc/ssh/ssh_config line 56: Applying options for *\
r\ndebug1: auto-mux: Trying existing master\r\ndebug2: fd 3 setting O_NONBLOCK\r\ndebug2: mux_client_hello_exchange: master version 4\r\ndebug3: mux_client_forwards: request forwardings: 0 local, 0 remote\r\ndebug3: mux_client_request_session: entering\r\ndebug3: mux_client_request_alive: entering\r\ndebug3: mux_client_request_alive: done pid = 4933\r\ndebug3: mux_client_request_session: session request sent\r\ndebug1: mux_client_request_session: master session id: 2\r\ndebug3: mux_client_read_packet: read header failed: Broken pipe\r\ndebug2: Received exit status from master 0\r\nShared connection to 127.0.0.1 closed.\r\n",     "module_stdout": "Traceback (most recent call last):\r\n  File \"/tmp/ansible_0abcTV/ansible_module_unarchive.py\", line 787, in <module>\r\n    main()\r\n  File \"/tmp/ansible_0abcTV/an
sible_module_unarchive.py\", line 747, in main\r\n    check_results = handler.is_unarchived()\r\n  File \"/tmp/ansible_0abcTV/ansible_module_unarchive.py\", line 416, in is_unarchived\r\n    if crc != self._crc32(path):\r\n  File \"/tmp/ansible_0abcTV/ansible_module_unarchive.py\", line 202, in _crc32\r\n    return self._infodict[path]\r\nKeyError: 'unarchive_test/#U65b0#U5efa#U6587#U672c#U6587#U6863.txt'\r\n",     "msg": "MODULE FAILURE", 
    "parsed": false
}
```

**Comment by ansibot** [comment]
@dagwieers, @pileofrogs, ping. This issue is waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by dagwieers** [comment]
This is caused by the fact that we enforce LANG=C so that scraping the output works cross-language.

If you comment out the line [#745](https://github.com/ansible/ansible-modules-core/blob/devel/files/unarchive.py#L745) you'll see it works fine:

``` python
    module.run_command_environ_update = dict(LANG='C', LC_ALL='C', LC_MESSAGES='C', LC_CTYPE='C')
```

@abadger Another reason to implement this pure-python and get it over with. :-(

wontfix as we plan to reimplement this as a pure-python **uncompress** module at some point in the future, so needs_contributor

**Comment by dagwieers** [comment]
@zhijieqing Can you please test PR #4702 , I think it may fix your problem.

**Comment by dagwieers** [comment]
resolved_by_pr #4702
